### PR TITLE
Add the call sc command client.

### LINKED ISSUE #2554: Add call smart contract to client

### PR DESCRIPTION
**Pull request by AurelienFT**
Fix #2554

### CODE REVIEW

**Review by Eitu33 — APPROVED** [review]

**Comment by AurelienFT** [comment]
bors merge

File: massa-client/src/cmds.rs [hunk 1]
```diff
@@ -815,6 +824,66 @@ impl Command {
...
+                        .and_then(|x| x.checked_add(coins))
+                        .and_then(|x| x.checked_add(fee))
+                    {
+                        Some(total) => {
+                            if let Ok(addresses_info) =
+                                client.public.get_addresses(vec![target_addr]).await
+                            {
+                                match addresses_info.get(0) {
+                                    Some(info) => {
+                                        if info.ledger_info.candidate_ledger_info.balance < total {
```
Comment (damip) [inline massa-client/src/cmds.rs:851]: note, in this case max_gas*gas_price+sequential_coins+fee are spent from the sequential balance, and parallel_coins are spent from the paralell balance ! so this check is invalid and  2 checks need to be performed

**Review by damip — CHANGES_REQUESTED** [review]

File: massa-client/src/cmds.rs (hunk 1 above)
Comment (AurelienFT) [reply massa-client/src/cmds.rs]: Is the new check ok ?

**Review by damip — APPROVED** [review]

**Comment by AurelienFT** [comment]
bors merge

**Pull request merged**